### PR TITLE
fix: customstatements in function events

### PR DESCRIPTION
**Pull request by praneetap**
*Issue #, if available:*

*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [ ] Write/update tests
- [ ] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected
- [ ] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by keetonian** [comment]
Verified that this fixes the issue we found.

Issue description: the previous fix (#1395) for resource policies introduced a regression where resource policies added in Function Events were not added to the resulting API. This fixes the regression.

File: examples/2016-10-31/api_resource_policy/template.yaml [hunk 1]
```diff
@@ -1,22 +1,33 @@
...
-                    "aws:SourceIp": "1.2.3.4"
-                }
-            }
-        }]
-        # OR you can use the following, they both do the same thing
-        IpRangeBlacklist: ['1.2.3.4']
+        CustomStatements:
+          Fn::If:
+            - C1
+            - Principal: '*'
```
Comment (jlhood) [inline examples/2016-10-31/api_resource_policy/template.yaml]: Why did you remove `Effect: Allow` from these statements?
